docs/geps/gep-07.md

```diff
@@ -0,0 +1,472 @@
+
+1. **Limited Flexibility**: The current interface makes it difficult to work with
+   different datasets / areas of the taxes and transfers system.
+
+This GEP aims to address these issues by introducing a more intuitive and flexible
+interface while maintaining GETTSIM's computational robustness.
+
+## Usage and Impact
+
+1. Basic workflow
+
+   There is a single entry point for GETTSIM: The `main` function. It is powered by a
+   DAG in the background.
+
+   This means that the user will have to start by telling it the desired target
+   ("`main_target`") or set of targets ("`main_targets`"). Ultimately, the main target
+   will typically be a dataset with values for taxes and transfers. However,
+   intermediate steps can be useful. E. g., the taxes and transfers system at a
+   particular date (the "`policy_environ­ment`"), which she wants to modify in order to
+   model a reform.
+
+   The targets determine the required inputs. For example, in order to compute taxes and
+   transfers for a set of households, one will need as primitives
+
+   - data on these households ("`input_data`")
+   - the set of taxes and transfers to be computed ("`tt_targets`"). This could be left
+     out, in which case all possible targets will be computed. However, that will often
+     be a daunting task in terms of requirements on the data and computer memory.
+   - the date
+
+   If only a policy environment is to be returned, just the date is required as an
+   input.
+
+1. **Worked example**
+
+   Here is an example (the variables `inputs_df`, `inputs_map`, and `targets_tree` will
+   be shown below).
+
+   ```python
+   from gettsim import InputData, MainTarget, TTTargets, main
+
+
+   outputs_df = main(
+       main_target=MainTarget.results.df_with_mapper,
+       date_str="2025-01-01",
+       input_data=InputData.df_and_mapper(
+           df=inputs_df,
+           mapper=inputs_map,
+       ),
+       tt_targets=TTTargets(tree=targets_tree),
+   )
+   ```
+
+   All elements that are not atomic are specified as GETTSIM objects, which means that
+   users can benefit from autocompletion and type hints provided by their IDE (see
+   below).
+
+   The first argument, `main_target`, specifies the type of object to compute. In this
+   case, we want the "results" in the "DataFrame with mapper" format. That is, GETTSIM
+   will compute all desired targets and return a DataFrame with columns specified by the
+   user.
+
+   Say we want to compute the contributions to long term care insurance
+   (Pflegeversicherung). The fourth argument, `tt_targets`, specifies the set of taxes
+   and transfers ("`tt`") to compute. Because we ask for the "results" in the "DataFrame
+   with mapper" format, this actually has to be a mapping from the targets to the
+   columns in the output DataFrame. In this case, the argument `tt_targets` needs to be
+   a *pytree*, which provides that mapping:
+
+   ```python
+   targets_tree = {
```

Comment:
How would the user typically come up with that? Look up the location of all targets she is interested in in a graph or in the code base directly?

Reply:
> same as before the DEFAULT_TARGETS, just different by date.

And that the "path" I need to look up and copy/type is substantially longer than the name of transfers before.

But I guess that's what we need to live with given the other advantages the new interface has.